src/components/Tasks/Tasks.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import { TasksContext } from '../../contexts/TasksContext'
+import { useTasks } from '../../hooks/useTasks'
+import { Container } from '../Container'
+import { Loader } from '../Loader'
+import { TasksHeader } from '../TasksHeader'
+import { TasksList } from '../TasksList'
+import { TasksPending } from '../TasksPending'
+
+export const Tasks = () => {
+  const contextData = useTasks()
+
+  return (
+    <TasksContext.Provider value={contextData}>
+      <Container name='tasks-component'>
+        <TasksHeader />
+        {contextData?.isLoading || !contextData.data ? (
```

Comment:
Not sure if you can do this that way... I think that ` {contextData?.isLoading ...}` is using data directly from `useTasks()` and not from `TasksContext.Provider`.

Let's split this into two components:

```ts
export const Tasks = () => {
  const contextData = useTasks()

  return (
    <TasksContext.Provider value={contextData}>
      <TasksComponent />
    </TasksContext.Provider>
  )
}

export const TasksComponent = () => {
  const contextData = useContext(TasksContext)

  return (
      <Container name='tasks-component'>
        <TasksHeader />
        {contextData?.isLoading || !contextData.data ? (
          <div className={`Layer__tasks__loader-container`}>
            <Loader />
          </div>
        ) : (
          <>
            {contextData.data.length > 0 && <TasksPending />}
            <TasksList />
          </>
        )}
      </Container>
  )
}
```